### PR TITLE
Adds specs for core classes

### PR DESCRIPTION
**Pull request by nolantait**
I was wanting to experiment with the library more and needed a test harness around the classes. Feel free to take what you want from here.

One spec is failing for the `Superform::NamespaceCollection#assign`. Not sure if this is the intended behavior or not but the namespace enumerator is altering the namespaces of objects that don't stick around between invocations of the enumerator. Could use some guidance here.

This PR also separates the classes out into files and formats for line length. I tried to leave the core code alone except for some small formatting changes. But you can always just use the spec folder.

I also added a development dependency for `debug`.

UPDATE: I figured out the namespace collection thing and pushed a fix to my spec. Turns out it was because of using doubles instead of an OpenStruct. I guess the collection namespace can't depend on the internal values of the fields for the reason I outlined above.

### CODE REVIEW

**Review by bradgessler — COMMENTED** [review]
I'd love to merge these tests, but all of the implementation files have changed. If you can clean up all the commits to the non-spec files it will be easier for me to review and merge.

File: Gemfile
Comment (bradgessler) [inline Gemfile:8]: I realize this might suck for your local development workflow, but I'd like to avoid adding new dependencies to this project. I'm hoping there's some way you can still use the `debug` gem for your local workflow or `binding.irb` is enough to get by. If none of those work and you plan on making lots of contributions then I'd consider merging in a separate commit and PR that explains the reason for this dependency to future committers.

File: lib/superform.rb
Comment (bradgessler) [inline lib/superform.rb:1]: Could this change be pulled out? Not a big deal if its a pain to do that.

**Review by bradgessler — CHANGES_REQUESTED** [review]
Can't wait to merge these tests! I'll merge if you're able to accommodate my feedback. Thanks!

File: lib/superform.rb
Comment (nolantait) [reply lib/superform.rb:1]: Yeah absolutely, its easy for local development to have the gem installed locally with `gem install debug` and then use the debugger without it being in the gemfile anyways. I've removed the dependency for you.

**Comment by bradgessler** [comment]
Yay tests! Thanks again for making these commits.

**Pull request merged**
